testing/upload_validation.ts

```diff
@@ -868,10 +872,57 @@ const genericObjectSchema: z.ZodTypeAny = z.lazy(() =>
           context.addIssue({
             code: z.ZodIssueCode.custom,
             path: ['featured', i],
-            message: `The featured field name "${f}" does not exist in the "properties" object.`,
+            message: `The "featured" field name "${f}" does not exist in the "properties" object.`,
           });
         }
       });
+    })
+    .superRefine((data, context) => {
+      const schemaHelper = objectSchemaHelper(data as GenericObjectSchema);
+
+      if (schemaHelper.imageProperty) {
+        if (!(schemaHelper.imageProperty in data.properties)) {
+          context.addIssue({
+            code: z.ZodIssueCode.custom,
+            path: ['imageProperty'],
+            message: `The "imageProperty" field name "${schemaHelper.imageProperty}" does not exist in the "properties" object.`,
+          });
+        }
+
+        // TODO: do the schema typing validation
+      }
+      if (schemaHelper.descriptionProperty) {
+        if (!(schemaHelper.descriptionProperty in data.properties)) {
+          context.addIssue({
+            code: z.ZodIssueCode.custom,
+            path: ['descriptionProperty'],
+            message: `The "descriptionProperty" field name "${schemaHelper.descriptionProperty}" does not exist in the "properties" object.`,
+          });
+        }
+
+        // TODO: do the schema typing validation
+      }
+      if (schemaHelper.linkProperty) {
+        if (!(schemaHelper.linkProperty in data.properties)) {
```

Comment:
taking a quick glance, it seems we never validate that `properties` is non-empty, might be nice to add

Reply:
good call. Do you also know why we break this validation up into a separate `superRefine` for each different validation? Was confused whether i should be bundling this here because it didnt seem to make a difference from the code

Reply:
Maybe it helps zod give better automated error messages describing where the problem exists.